src/alchemlyb/tests/test_visualisation.py

```diff
@@ -1,4 +1,5 @@
 import matplotlib
+matplotlib.use('Agg')
```

Comment:
Instead of hard coding in the test, set an environment variable in the ci.yml file
```yaml
    env:
      MPLBACKEND: agg
```

Reply:
P.S.: Good that you identified the issue!